integration/health_test.go

```diff
@@ -95,6 +97,158 @@ func TestAddHealthCheckSources(t *testing.T) {
+			return nil, nil
+		},
+		ioutil.Discard,
+		func(t *testing.T, initFn witchcraft.InitFunc, installCfg config.Install, logOutputBuffer io.Writer) *witchcraft.Server {
+			return createTestServer(t, initFn, installCfg, logOutputBuffer)
+		},
+	)
+
+	getHealth := func() health.HealthStatus {
+		resp, err := testServerClient().Get(fmt.Sprintf("https://localhost:%d/%s/%s", port, basePath, status.HealthEndpoint))
+		require.NoError(t, err)
+
+		bytes, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+
+		var healthResults health.HealthStatus
+		err = json.Unmarshal(bytes, &healthResults)
+		require.NoError(t, err)
+		return healthResults
+	}
+
+	defer func() {
+		require.NoError(t, server.Close())
+	}()
+	defer cleanup()
+
+	okServer := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.WriteHeader(http.StatusOK)
+	}))
+	defer okServer.Close()
+	errServer := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.WriteHeader(http.StatusServiceUnavailable)
+	}))
+	defer errServer.Close()
+	errHost := strings.TrimPrefix(errServer.URL, "http://")
+	stoppedServer := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.WriteHeader(http.StatusServiceUnavailable)
+	}))
+	stoppedServer.Close()
+	stoppedHost := strings.TrimPrefix(stoppedServer.URL, "http://")
+
+	assert.Equal(t, health.HealthStatus{
+		Checks: map[health.CheckType]health.HealthCheckResult{
+			health.CheckType("CONFIG_RELOAD"): {
+				Type:   health.CheckType("CONFIG_RELOAD"),
+				State:  health.New_HealthState(health.HealthState_HEALTHY),
+				Params: make(map[string]interface{}),
+			},
+			health.CheckType("SERVER_STATUS"): {
+				Type:   health.CheckType("SERVER_STATUS"),
+				State:  health.New_HealthState(health.HealthState_HEALTHY),
+				Params: make(map[string]interface{}),
+			},
+		},
+	}, getHealth())
+
+	clientA, err := clients.NewHTTPClient(ctx, "serviceA")
+	require.NoError(t, err)
+	_, _ = clientA.CurrentHTTPClient().Get(okServer.URL)
+
+	require.Equal(t, health.HealthStatus{
+		Checks: map[health.CheckType]health.HealthCheckResult{
+			health.CheckType("CONFIG_RELOAD"): {
+				Type:   health.CheckType("CONFIG_RELOAD"),
+				State:  health.New_HealthState(health.HealthState_HEALTHY),
+				Params: make(map[string]interface{}),
+			},
+			health.CheckType("SERVER_STATUS"): {
+				Type:   health.CheckType("SERVER_STATUS"),
+				State:  health.New_HealthState(health.HealthState_HEALTHY),
+				Params: make(map[string]interface{}),
+			},
+			health.CheckType("SERVICE_DEPENDENCY"): {
+				Type:    health.CheckType("SERVICE_DEPENDENCY"),
+				Message: stringPtr("All remote services are healthy"),
+				State:   health.New_HealthState(health.HealthState_HEALTHY),
+				Params:  make(map[string]interface{}),
+			},
+		},
+	}, getHealth())
+
+	_, _ = clientA.CurrentHTTPClient().Get(errServer.URL)
+	_, _ = clientA.CurrentHTTPClient().Get(errServer.URL)
```

Comment:
Should we just set the `httpclient.WithMaxRetries(0)` rather than request twice? I don't have strong opinions either way, but this could be confusing if you don't know that CGR retries 2 times by default.